runtime/lua/vim/health/health.lua

```diff
@@ -493,11 +493,112 @@ local function check_external_tools()
   end
 end
 
+local function detect_terminal()
+  local e = vim.env
+  if e.TERM_PROGRAM then
+    local v = e.TERM_PROGRAM_VERSION --- @type string?
+    return e.TERM_PROGRAM_VERSION and (e.TERM_PROGRAM .. ' ' .. v) or e.TERM_PROGRAM
+  end
+
+  local map = {
+    KITTY_WINDOW_ID = 'kitty',
+    ALACRITTY_SOCKET = 'alacritty',
+    ALACRITTY_LOG = 'alacritty',
+    WEZTERM_EXECUTABLE = 'wezterm',
+    KONSOLE_VERSION = function()
+      return 'konsole ' .. e.KONSOLE_VERSION
+    end,
+    VTE_VERSION = function()
+      return 'vte ' .. e.VTE_VERSION
+    end,
+  }
+
+  for key, val in pairs(map) do
+    local env = e[key] --- @type string?
+    if env then
+      return type(val) == 'function' and val() or val
+    end
+  end
+
+  return 'unknown'
+end
+
+local function check_sysinfo()
+  vim.health.start('Bug Report Information')
+
+  -- Use :version because vim.version().build returns "Homebrew" for brew installs.
+  local version_output = vim.api.nvim_exec2('version', { output = true }).output
+  local nvim_version = version_output:match('NVIM v[^\n]+') or 'unknown'
+  local commit_hash = nvim_version:match('%+g?(%x+)')
+  local version_for_report = nvim_version
+  if commit_hash then
+    version_for_report = nvim_version:gsub('%+g' .. commit_hash, ' neovim/neovim@' .. commit_hash)
+    local result = vim
+      .system({ 'git', 'ls-remote', 'https://github.com/neovim/neovim', 'HEAD' }, { text = true })
+      :wait()
```

Comment:
what happens if `git` is not on the system? checkhealth shouldn't just fail. this will raise an error.